lib/api/initiateMultipartUpload.js

```diff
@@ -164,6 +169,13 @@ function initiateMultipartUpload(authInfo, request, log, callback) {
         const serverSideEncryption =
             destinationBucket.getServerSideEncryption();
         let cipherBundle = null;
+        if ((!serverSideEncryption && sseHeader) ||
+            (sseHeader === 'AES256' &&
```

Comment:
It seems that it will not return any error if bucket encryption is enabled and `sseHeader` is not equal `'AES256'`

Reply:
I think it will, because in that case the `serverSideEncryption.algorithm !== sseHeader` condition would be true

Reply:
I am probably missing something out but if `serverSideEncryption.algorithm = 'AES256' ` and `sseHeader = 'invalidheader'` would it not end up `(false && true ) || (false && true)` ?

Reply:
oh yeah, you right. That condition should just check if `sseHeader` exists